python/pyarrow/_flight.pyx

```diff
@@ -704,7 +704,7 @@ cdef class FlightEndpoint(_Weakrefable):
     cdef:
         CFlightEndpoint endpoint
 
-    def __init__(self, ticket, locations):
+    def __init__(self, ticket, locations, expiration_time=None, app_metadata=""):
```

Comment:
```suggestion
    def __init__(self, ticket, locations, TimestampScalar expiration_time=None, app_metadata=""):
```

If we require such a scalar, I would either enforce that through cython in the signature, or otherwise add a check in the code below that ensures a proper values is passed. For example, right now if you pass a stdlib datetime, you get "'datetime.datetime' object has no attribute 'cast'"

Reply:
Added type checks in 415aca76b. Note none of the other arguments where checked, so I added that for all arguments. This includes tests.